src/cdk/datepicker/datepicker-input.ts

```diff
@@ -0,0 +1,363 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {coerceBooleanProperty} from '@angular/cdk/coercion';
+import {
+  AfterContentInit,
+  Directive,
+  ElementRef,
+  EventEmitter,
+  forwardRef,
+  Inject,
+  Input,
+  OnDestroy,
+  Output,
+} from '@angular/core';
+import {
+  AbstractControl,
+  ControlValueAccessor,
+  NG_VALIDATORS,
+  NG_VALUE_ACCESSOR,
+  ValidationErrors,
+  Validator,
+  ValidatorFn,
+  Validators
+} from '@angular/forms';
+import {CdkDatepicker} from './datepicker';
+import {DateAdapter, MAT_DATE_FORMATS, MatDateFormats} from '@angular/material/core';
+import {MatFormField} from '@angular/material/form-field';
+import {MAT_INPUT_VALUE_ACCESSOR} from '@angular/material/input';
+import {Subscription} from 'rxjs';
+
+export const CDK_DATEPICKER_VALUE_ACCESSOR: any = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => CdkDatepickerInput),
+  multi: true
+};
+
+
+export const CDK_DATEPICKER_VALIDATORS: any = {
+  provide: NG_VALIDATORS,
+  useExisting: forwardRef(() => CdkDatepickerInput),
+  multi: true
+};
+
+
+/**
+ * An event used for datepicker input and change events. For consistency, we always use
+ * CdkDatepickerInputEvent instead.
+ */
+export class CdkDatepickerInputEvent<D> {
```

Comment:
We are using the constructor of `CdkDatepickerInputEvent` in the `emitDateInput` and `emitDateChange` functions, where the events are emitted on the input, and a new input event is created. This requires that the value of the datepicker input that emitted the event should be assigned to the value of the input event. This applies to the `MatDatepickerInputEvent<D>` as well.

Reply:
Is there another approach that you were thinking of?

Reply:
Are you suggesting that because we have access to a native input or change event, the constructor isn't needed and instead, we can use the `HTMLInputElement`? I'll make the change and see how to proceed from there.